packages/tools/devtools/devtools-view/src/components/graphs/DynamicComposedChart.tsx

```diff
@@ -0,0 +1,394 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// Recharts does not have types for many objects at this time. For now, these eslint disable directive should be active until we create our own types for recharts
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { Theme } from "@fluentui/react-components";
+import React, { useState } from "react";
+import {
+	Area,
+	Bar,
+	CartesianGrid,
+	ComposedChart,
+	Legend,
+	Line,
+	ResponsiveContainer,
+	Tooltip,
+	XAxis,
+	YAxis,
+} from "recharts";
+import { ThemeContext } from "../../ThemeHelper";
+
+/**
+ * Data To be rendered with Op Latency Graph
+ */
+export interface GraphDataSet {
+	graphType: "line" | "area" | "bar";
+	schema: {
+		displayName: string;
+		uuid: string;
+		xAxisDataKey: string;
+		yAxisDataKey: string;
```

Comment:
I'm trying to think if we gain something important by allowing the consumer to specify these and providing an array of objects, vs just taking in an array of 2-tuples. Did you run into particular scenarios where one or the other has any advantage?

Reply:
My original thought here was to provide two things, convenience and performance.
 - Convenience because consumers don't have to transform the attribute names of the data they are passing.
 - Performance because we skip one data transformation step.

However, I acknowledge that it is unusual and that it might be a premature performance tweak. I'm happy to make this more simplistic since I haven't exactly verified the performance increase and it may be negligible for our use case

Reply:
It all depends on how the source data is structured. We're the only consumers so we'll be in control of that, and we have a decent idea of how frequently we'll have to go (or not) through this step (not thousands of times a second, for sure) so I'm not particularly concerned about performance one way or the other. Don't change it now, but maybe we revisit in the future.